### ISSUE TITLE
Incorrect order of elements in sstables

### ISSUE THREAD

**Issue by pdziepak** [scylladb/scylla#1203]
Scylla version (or git commit hash): 2e08d0f698090f978f5a94e3ae4fc73f0d8c6b5b

For each partition in an sstable Scylla writes all range tombstones first and only then clustering rows. As a result such sstables aren't really sorted.

For example:
```
create table cf1 (p int, a int, b int, c int, r int, primary key (p,a,b,c)) with compact storage;
insert into cf1 (p,a,b,c,r) values (0,1,2,3,4);
insert into cf1 (p,a,b,r) values (0,1,2,5);
insert into cf1 (p,a,r) values (0,1,5);
delete from cf1 where p = 0 and a = 1 and b = 3;
insert into cf1 (p,a,r) values (0,1,6);
insert into cf1 (p,a,b,c,r) values (0,1,3,4,7);
delete from cf1 where p = 0 and a = 1 and b = 3;
insert into cf1 (p,a,b,r) values (0,1,3,8);
```

Cassandra 2.1.13:
```
[
{"key": "0",
 "cells": [["1","6",1461056290485699],
           ["1:2","5",1461056290478406],
           ["1:2:3","4",1461056290469635],
           ["1:3:_","1:3:!",1461056290493505,"t",1461056290],
           ["1:3","8",1461056290496832]]}
]
```

Scylla:
```
[
{"key": "0",
 "cells": [["1:3:_","1:3:!",1461056427956199,"t",1461056427],
           ["1","6",1461056427949214],
           ["1:2","5",1461056427939833],
           ["1:2:3","4",1461056427934309],
           ["1:3","8",1461056427959635],
           ["1:3:4","7",1461056427952646]]}
]
```

**Comment by slivne** [comment]
Pawel,

What is the effect of this on Scylla right now - after chatting with Nadav
- we are reading a complete partition - so we will read all the data in any
case (no matter what the order is).

When we will do streaming this is off course problematic.

There may be an issue with taking a Scylla file and moving it back to
Cassandra - that Cassandra may not read correctly.

**Comment by slivne** [comment]
I don't think so -

**Comment by pdziepak** [comment]
I don't think streaming is going to be a problem. It sends frozen mutations, so the way sstables are written is irrelevant.

The problem indeed is with using Scylla sstables with Cassandra.. I only did a quick experiment and moved the Scylla-generated sstable from this issue to Cassandra. At first Cassandra was able to read it correctly – select produced correct result (I haven't checked the code so I don't know if it will be able to read all Scylla-generated sstables correctly). However, after I forced compaction Cassandra dropped all rows between the misplaced range tombstone and the first row at the correct position, the result is:
```
[
{"key": "0",
 "cells": [["1:3:_","1:3:!",1461056427956199,"t",1461056427],
           ["1:3","8",1461056427959635]]}
]
```
